packages/rollup-full-node/src/app/web3-rpc-handler.ts

```diff
@@ -116,70 +117,72 @@ export class DefaultWeb3Handler
 
     // Make sure the method is available
     let response: any
-    let args: any[]
     switch (method) {
       case Web3RpcMethods.blockNumber:
-        this.assertParameters(params, 0)
+        this.assertParameters(params, [])
         response = await this.blockNumber()
         break
       case Web3RpcMethods.call:
-        args = this.assertParameters(params, 2, latestBlock)
-        response = await this.call(args[0], args[1])
+        this.assertParameters(params, ['object', 'quantity|tag'])
```

Comment:
Ohh, I like this! 

Might suggest an enum for types instead of string literals. If you make their values powers of two, bitwise operations will work seamlessly.

Reply:
[Converted to enums](https://github.com/ethereum-optimism/optimism-monorepo/pull/115/commits/720907cc09827bc1d1f31d9b346940fc0c6fc799) 👍 

It'd be awesome if we could make use of bitwise oring but I think that'd be a significant amount of work without much upside.

We'd have to pass "quantity | tag" into assertParameters as a parameter which you could do as either a bitmask or an array. Then we'd have to have branch in assertParameters and yada yada yada. Also there's only one case where things are ored together and very little chance others will be added. Happy to chat about it if you've got other ideas and wanna hop on a call though! 😀

Reply:
That works! Didn't know how composable these would need to be. Looks like not very. This is a good pragmatic approach 👍